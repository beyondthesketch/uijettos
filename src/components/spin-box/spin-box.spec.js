import React from 'react';
import renderer from 'react-test-renderer';

import UijettosSpinBox from './spin-box';

describe('UijettosSpinBox component', () => {
    test('Renders correctly when not supplied any props', () => {
        const tree = renderer.create(
            <UijettosSpinBox />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when supplied with an initialValue', () => {
        const tree = renderer.create(
            <UijettosSpinBox
                initialValue={ 27 }
            />
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when supplied with a cssClassPrefix', () => {
        const tree = renderer.create(
            <UijettosSpinBox
                cssClassPrefix="widget"
                initialValue={ 27 }
            />
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when supplied with a min value', () => {
        const tree = renderer.create(
            <UijettosSpinBox
                min={ 27 }
            />
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when supplied with a max value', () => {
        const tree = renderer.create(
            <UijettosSpinBox
                max={ 27 }
            />
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when supplied with a step value', () => {
        const tree = renderer.create(
            <UijettosSpinBox
                step={ 3 }
            />
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when supplied with autoWidth as true', () => {
        const tree = renderer.create(
            <UijettosSpinBox
                autoWidth={ true }
            />
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    });
});