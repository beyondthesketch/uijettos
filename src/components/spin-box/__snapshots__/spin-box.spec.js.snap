// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UijettosSpinBox component Renders correctly when not supplied any props 1`] = `
<div
  className="uijettos-spin-box"
>
  <input
    className="uijettos-spin-box__input"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    step={1}
    style={null}
    type="number"
    value="0"
  />
  <span
    className="uijettos-spin-box__button uijettos-spin-box__button--decrease"
    onClick={[Function]}
  >
    Decrease
  </span>
  <span
    className="uijettos-spin-box__button uijettos-spin-box__button--increase"
    onClick={[Function]}
  >
    Increase
  </span>
</div>
`;

exports[`UijettosSpinBox component Renders correctly when supplied with a cssClassPrefix 1`] = `
<div
  className="widget-spin-box"
>
  <input
    className="widget-spin-box__input"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    step={1}
    style={null}
    type="number"
    value="27"
  />
  <span
    className="widget-spin-box__button widget-spin-box__button--decrease"
    onClick={[Function]}
  >
    Decrease
  </span>
  <span
    className="widget-spin-box__button widget-spin-box__button--increase"
    onClick={[Function]}
  >
    Increase
  </span>
</div>
`;

exports[`UijettosSpinBox component Renders correctly when supplied with a max value 1`] = `
<div
  className="uijettos-spin-box"
>
  <input
    className="uijettos-spin-box__input"
    max={27}
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    step={1}
    style={null}
    type="number"
    value="0"
  />
  <span
    className="uijettos-spin-box__button uijettos-spin-box__button--decrease"
    onClick={[Function]}
  >
    Decrease
  </span>
  <span
    className="uijettos-spin-box__button uijettos-spin-box__button--increase"
    onClick={[Function]}
  >
    Increase
  </span>
</div>
`;

exports[`UijettosSpinBox component Renders correctly when supplied with a min value 1`] = `
<div
  className="uijettos-spin-box"
>
  <input
    className="uijettos-spin-box__input"
    min={27}
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    step={1}
    style={null}
    type="number"
    value="0"
  />
  <span
    className="uijettos-spin-box__button uijettos-spin-box__button--decrease uijettos-spin-box__button--disabled"
    onClick={[Function]}
  >
    Decrease
  </span>
  <span
    className="uijettos-spin-box__button uijettos-spin-box__button--increase"
    onClick={[Function]}
  >
    Increase
  </span>
</div>
`;

exports[`UijettosSpinBox component Renders correctly when supplied with a step value 1`] = `
<div
  className="uijettos-spin-box"
>
  <input
    className="uijettos-spin-box__input"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    step={3}
    style={null}
    type="number"
    value="0"
  />
  <span
    className="uijettos-spin-box__button uijettos-spin-box__button--decrease"
    onClick={[Function]}
  >
    Decrease
  </span>
  <span
    className="uijettos-spin-box__button uijettos-spin-box__button--increase"
    onClick={[Function]}
  >
    Increase
  </span>
</div>
`;

exports[`UijettosSpinBox component Renders correctly when supplied with an initialValue 1`] = `
<div
  className="uijettos-spin-box"
>
  <input
    className="uijettos-spin-box__input"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    step={1}
    style={null}
    type="number"
    value="27"
  />
  <span
    className="uijettos-spin-box__button uijettos-spin-box__button--decrease"
    onClick={[Function]}
  >
    Decrease
  </span>
  <span
    className="uijettos-spin-box__button uijettos-spin-box__button--increase"
    onClick={[Function]}
  >
    Increase
  </span>
</div>
`;

exports[`UijettosSpinBox component Renders correctly when supplied with autoWidth as true 1`] = `
<div
  className="uijettos-spin-box"
>
  <input
    className="uijettos-spin-box__input"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    step={1}
    style={
      Object {
        "minWidth": "2ch",
        "width": "2ch",
      }
    }
    type="number"
    value="0"
  />
  <span
    className="uijettos-spin-box__button uijettos-spin-box__button--decrease"
    onClick={[Function]}
  >
    Decrease
  </span>
  <span
    className="uijettos-spin-box__button uijettos-spin-box__button--increase"
    onClick={[Function]}
  >
    Increase
  </span>
</div>
`;
